src/nitypes/scalar/_scalar.py

```diff
@@ -0,0 +1,145 @@
+from __future__ import annotations
+
+from typing import Generic, Union
+
+from typing_extensions import final, TypeVar
+
+from nitypes._exceptions import invalid_arg_type, invalid_arg_value
+from nitypes.waveform._extended_properties import (
+    UNIT_DESCRIPTION,
+    ExtendedPropertyDictionary,
+)
+
+_ScalarType_co = TypeVar("_ScalarType_co", bound=Union[bool, int, float, str], covariant=True)
+_NUMERIC = (bool, int, float)
+
+
+@final
+class Scalar(Generic[_ScalarType_co]):
+    """A scalar data class, which encapsulates scalar data and units information."""
+
+    __slots__ = [
+        "_value",
+        "_extended_properties",
+    ]
+
+    _value: _ScalarType_co
+    _extended_properties: ExtendedPropertyDictionary
+
+    def __init__(
+        self,
+        value: _ScalarType_co,
+        units: str = "",
+    ) -> None:
+        """Construct a scalar data object.
```

Comment:
```suggestion
        """Initialize a new Scalar data object.
```

I suggested "Initialize" because the dunder method `__new__()` does the creation.